Public/Sdk/Public/Prelude/Prelude.Configuration.Resolvers.dsc

```diff
@@ -136,12 +136,28 @@ interface MsBuildResolver extends ResolverBase, UntrackingSettings {
     runInContainer?: boolean;
 
     /**
-     * Collection of directories to search for the required MsBuild assemblies and MsBuild.exe (a.k.a. MSBuild toolset).
+     * Collection of directories to search for the required MsBuild assemblies and MsBuild.exe/MSBuild.dll (a.k.a. MSBuild toolset).
      * If not specified, locations in %PATH% are used.
      * Locations are traversed in specification order.
     */
     msBuildSearchLocations?: Directory[];
 
+    /**
+     * Whether to use the full framework or dotnet core version of MSBuild. Selected runtime is used both for build evaluation and execution.
+     * Default is full framework.
+     * Observe that using the full framework version means that msbuild.exe is expected to be found in msbuildSearchLocations 
+     * (or PATH if not specified). If using the dotnet core version, the same logic applies but to msbuild.dll
+     */
+    msBuildRuntime?: "FullFramework" | "DotNetCore";
+
+    /**
+     * Collection of directories to search for dotnet.exe, when DotNetCore is specified as the msBuildRuntime. If not 
+     * specified, locations in %PATH% are used.
+     * Locations are traversed in specification order.
+     * It has no effect if the specified MSBuild runtime is full framework.
+     */
+    dotNetSearchLocations?: Directory[];

```

Comment:
just curious: why couldn't you overload and use the same `msBuildSearchLocations` property for this?

Reply:
Not a strong reason... I just thought it was cleaner this way, the locations probably won't match anyway

---
In reply to: [301280165](https://github.com/microsoft/BuildXL/pull/561#discussion_r301280165) [](ancestors = 301280165)